src/java.base/share/classes/java/io/Reader.java

```diff
@@ -391,6 +395,40 @@ public int read(char[] cbuf) throws IOException {
      */
     public abstract int read(char[] cbuf, int off, int len) throws IOException;
 
+    /**
+     * Reads all remaining characters into a string.
+     *
+     * <p> This method reads all content including the line separators in
```

Comment:
Again, just to say that it's all remaining rather than "all content" as some characters may have been consumed before this method is invoked.

Now that you have readAllChars, I think it would be good to prototype a readAllLines (or better name) on Reader too. Consuming all characters means that Reader should not need to take on the characteristics of a buffered-Reader.

Reply:
And more: 1bbb7b1.